force-app/main/default/classes/RD2_RecurringDonation.cls

```diff
@@ -494,6 +528,15 @@ public inherited sharing class RD2_RecurringDonation {
         return rd.npe03__Installment_Period__c;
     }
 
+    public Date getExpectedDonationDate(RD2_ScheduleService scheduleService) {
+        Date baseDate = (rd.npe03__Last_Payment_Date__c == null) ? rd.StartDate__c : rd.npe03__Last_Payment_Date__c;
+        return scheduleService.getNextDonationDateAfter(
+            baseDate.addDays(1),
```

Comment:
When a new RD's startDate is same as the Donation date, do we want to give 1 extra installment window before marking the RD is Lapsed/Closed. For example, a monthly RD with day of month = 1, and the RD is created on the 2/1/2022. This will result the first donation to be on 2/1/2022. With this setting, the RD will be mark as Lasped/Closed after 3/1/2022 + number of days for Lapsed/Closed (ignore the 2/1/2022 donation).